lib/pages/chat_list/chat_list.dart

```diff
@@ -170,18 +172,84 @@ class ChatListController extends State<ChatList>
 
   List<String> getBotIds() {
     final hostName = getHostName();
-    return SocialNetworkManager.socialNetworks.map((sn) => sn.chatBot + hostName).toList();
+    return SocialNetworkManager.socialNetworks
+        .map((sn) => sn.chatBot + hostName)
+        .toList();
   }
 
   // List of user or bot IDs to exclude
-  List<String> get excludedUserIds => getBotIds();
+  List<String> get excludedUserIds => _excludedUserIds;
+
+  bool isGroupWithOnlyBotAndUser(Room room) {
+    final client = Matrix.of(context).client;
+    final participants = room.getParticipants();
+    return participants.length == 2 &&
+        participants.any((user) => _excludedUserIds.contains(user.id)) &&
+        participants.any((user) => user.id == client.id.toString());
+  }
+
+  // Method to identify and remove duplicate rooms
+  void identifyAndRemoveDuplicates(List<Room> rooms) {
+    final Map<String, List<Room>> roomMap = {};
+    final client = Matrix.of(context).client;
+
+    // Group rooms by the bot ID in _excludedUserIds
+    for (var room in rooms) {
+      String? botId = room.directChatMatrixID;
+      if (botId == null && isGroupWithOnlyBotAndUser(room)) {
+        botId = _excludedUserIds.firstWhere(
+              (id) => room.getParticipants().any((user) => user.id == id),
+          orElse: () => '',
```

Comment:
code would be cleaner if `orElse` resolved to `null`

Reply:
casting them all to a string defeats the purpose. My bad, the original was good :+1: